client/blocks/login/social.jsx

```diff
@@ -28,13 +28,15 @@ import { InfoNotice } from 'blocks/global-notice';
 import { login } from 'lib/paths';
 
 function isSafari() {
-	return typeof window !== 'undefined' && window.navigator.userAgent.match( /safari/i );
+	return (
+		typeof window !== 'undefined' && /^(?!.*chrome).*safari/i.test( window.navigator.userAgent )
+	);
 }
 
 function shouldUseRedirectFlow() {
 	// If calypso is loaded in a popup, we don't want to open a second popup for social login
 	// let's use the redirect flow instead in that case
-	const isPopup = typeof window !== 'undefined' && window.opener && window.opener !== window;
+	const isPopup = typeof window !== 'undefined' && window.opener !== window;
```

Comment:
<img width="622" alt="screen shot 2017-10-11 at 0 55 28" src="https://user-images.githubusercontent.com/326402/31412733-f35a973c-ae1e-11e7-9b35-e560897d0b80.png">

I think it should remain `&& window.opener &&`

Reply:
Right 👍